src/message.rs

```diff
@@ -53,6 +53,32 @@ impl ZmqMessage {
     pub fn into_vecdeque(self) -> VecDeque<Bytes> {
         self.frames
     }
+
+    pub fn prepend(&mut self, message: &ZmqMessage) {
+        for frame in message.iter().rev() {
+            self.push_front(frame.clone());
+        }
+    }
+
+    pub fn split_at(&self, mid: usize) -> (ZmqMessage, ZmqMessage) {
```

Comment:
You could use split_off - https://doc.rust-lang.org/std/collections/struct.VecDeque.html#method.split_off